src/coprocessor/endpoint.rs

```diff
@@ -602,11 +602,23 @@ impl<E: Engine> Endpoint<E> {
     }
 }
 
+fn is_engine_corruption_error(e: &Error) -> bool {
+    // Hack! it depend on the error message to identify the engine corruption error
+    let err_msg = e.to_string().to_lowercase();
```

Comment:
what's the purpose

Reply:
The `engine_traits::Error` is set to the `Error::message` field.